src/sage/algebras/lie_algebras/superliealgebra.py

```diff
@@ -0,0 +1,354 @@
+                if test_bk != bk:
+                    raise ValueError("contradictory linear system")
+
+            return [sum((R(Aij) * bk for Aij,bk in zip(Ai,b)), cm.zero())
+                    for Ai in A_inv.rows()]
+
+        bracketlength = 1
+        n = 0
+        while True:
+            sm = m.submodule(spanning_set)
+            A = matrix(sm.base_ring(), [sm.coordinate_vector(X) for X in spanning_set])
+            try:
+                im_gens = solve_linear_system(A, im_gens)
+            except ValueError:
+                raise ValueError("this does not define a graded Lie algebra morphism")
+
+            spanning_set = list(sm.basis())
+            if n == len(spanning_set):
+
+                break
+
+            bracketlength += 1
+            n = len(spanning_set)
+            for i,j in combinations(range(n), 2):
+
+                Z = list(on_generators.keys())[i].bracket(list(on_generators.keys())[j])
+                imZ = list(on_generators.values())[i].bracket(list(on_generators.values())[j])
+                spanning_set.append(Z.to_vector())
+                im_gens.append(imZ.to_vector())
+
+        A = matrix(m.base_ring(), spanning_set)
+        im_gens = solve_linear_system(A, im_gens)
+
+        bh = lambda t: t
+        return cm.sum(bh(c) * im_gens[i] for i, c in (argument.to_vector()).items())
+
+
+    class Element(CombinatorialFreeModule.Element):
+         
+        #Returns the degree of the highest-degree monomial in any expression:
+        def degree(self):
+
+            l = [m for m in self]
+            monomials = [x[0] for x in l]
+            dictionary = self.parent().name_degree_map
+            degrees=self.parent()._degrees
+
+            values = [dictionary[key] for key in monomials if key in dictionary]
+            if values:
+                return max(values)
+            else:
+                return 0
+            
+        #Returns the degree of a homogeneous expression:
+        def homdegree(self):
+
+            l = [m for m in self]
+            monomials = [x[0] for x in l]
+            dictionary = self.parent().name_degree_map
+            degrees=self.parent()._degrees
+
+            values = [dictionary[key] for key in monomials if key in dictionary]
+            if values:
+                if len(values)!=1:
+                    return "Not homogeneous"
+                else:
+                    return values[0]
+            else:
+                return 0
+
+        def bracket(self, rt):
+            names=self.parent()._names
+            degrees=self.parent()._degrees
+            dictionary = self.parent().vector_presentation
+            basis=self.parent().basis()
+
+            structurecoefficients = self.parent()._s_coeff
+            d = self.parent().dimension()
+            ret = [0]*d
+
+            if self == 0:
+                return 0
+            if rt == 0:
+                return 0
```

Comment:
```suggestion
            if not self or not rt:
                return P.zero()
```
This is faster and you need to return an element of the Lie algebra.